test/integration/filters/decode_headers_return_stop_all_filter.cc

```diff
@@ -0,0 +1,92 @@
+#include <chrono>
+#include <string>
+
+#include "envoy/event/timer.h"
+#include "envoy/http/filter.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "common/buffer/buffer_impl.h"
+
+#include "extensions/filters/http/common/empty_http_filter_config.h"
+#include "extensions/filters/http/common/pass_through_filter.h"
+
+#include "test/integration/filters/common.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+// A filter returns StopAllIterationAndBuffer or StopAllIterationAndWatermark for headers. The
+// iteration continues after 5s.
```

Comment:
I'm a bit wary of adding a bunch of 5s pauses to protocol integration test.
Is the goal here to wait until all the bytes are read?  Could we have a fast alarm which checked stream info bytesReceived() and triggered the onData once they'd all been read?

Reply:
Good idea! Fixed.